.buildkite/packaging.pipeline.yml

```diff
@@ -0,0 +1,229 @@
+        agents:
+          provider: gcp
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "c2-standard-16"
+        artifact_paths:
+          - build/distributions/**/*
+        matrix:
+          - x-pack/agentbeat
+          - x-pack/osquerybeat
+
+  - group: Packaging staging
+
+    key: packaging-staging
+    ## Only for release and main
+    if: "build.branch =~ /\\d+\\.\\d+/ || build.branch == 'main'"
+    steps:
+      # x86
+      - label: ":ubuntu: {{matrix}}/Packaging Linux Staging"
+        env:
+          PLATFORMS: "+all linux/amd64 linux/arm64 windows/amd64 darwin/amd64 darwin/arm64"
+          SNAPSHOT: false
+        command: ".buildkite/scripts/packaging/package-dra.sh {{matrix}}"
+        agents:
+          provider: gcp
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "${GCP_HI_PERF_MACHINE_TYPE}"
+        artifact_paths:
+          - build/distributions/**/*
+        matrix:
+          - auditbeat
+          - filebeat
+          - heartbeat
+          - metricbeat
+          - packetbeat
+          - winlogbeat
+          - x-pack/auditbeat
+          - x-pack/dockerlogbeat
+          - x-pack/filebeat
+          - x-pack/functionbeat
+          - x-pack/heartbeat
+          - x-pack/metricbeat
+          - x-pack/packetbeat
+          - x-pack/winlogbeat
+
+      ## ARM
+      - label: ":linux: {{matrix}}/Packaging Linux arm64 Staging"
+        env:
+          PLATFORMS: "linux/arm64"
+          PACKAGES: "docker"
+          SNAPSHOT: false
+        command: ".buildkite/scripts/packaging/package-dra.sh {{matrix}}"
+        agents:
+          provider: "aws"
+          imagePrefix: "${AWS_IMAGE_UBUNTU_ARM_64}"
+          instanceType: "${AWS_ARM_INSTANCE_TYPE}"
+        artifact_paths:
+          - build/distributions/**
+        matrix:
+          - auditbeat
+          - filebeat
+          - heartbeat
+          - metricbeat
+          - packetbeat
+          - x-pack/auditbeat
+          - x-pack/dockerlogbeat
+          - x-pack/filebeat
+          - x-pack/heartbeat
+          - x-pack/metricbeat
+          - x-pack/packetbeat
+
+        ## Osquerybeat and agentbeat require msitools to be installed
+        ## TODO: pre-install msitools to the VM and delete this exclusion
+      - label: ":ubuntu: {{matrix}}/Packaging Linux Staging"
+        env:
+          PLATFORMS: "+all linux/amd64 linux/arm64 windows/amd64 darwin/amd64 darwin/arm64"
+          SNAPSHOT: false
+        command: ".buildkite/scripts/packaging/package-dra.sh {{matrix}}"
+        agents:
+          provider: gcp
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "${GCP_HI_PERF_MACHINE_TYPE}"
+        artifact_paths:
+          - build/distributions/**
+        matrix:
+          - x-pack/agentbeat
+          - x-pack/osquerybeat
+
+  - group: DRA publish
+    key: dra
+    steps:
+      - label: DRA Snapshot
+        key: dra-snapshot
+        env:
```

Comment:
the snapshot is something that we are running for the release branch and the main branch, so this step should have the following condition
```yaml
if: "build.branch =~ /\\d+\\.\\d+/ || build.branch == 'main'"
```

Additionally, having an env var to loose the strictness, similar to the staging step bellow, would be good.

Reply:
Updated conditions